### PR TITLE
Save seed data in components and remove the seed-database

### PR DESCRIPTION
**Pull request by ElectroJr**
This PR removes the botany system's seed database, in favour of just storing seed data on relevant components. The main advantage is that non-standard seeds can now actually be saved to a map.

The `SeedPrototype` class was split into a `SeedData` class and a prototype class that inherits from that. Whenever a seed is about to be modified, the botany system should either ensure that the seed data is either unique, or it should create a new clone of the existing data.

The seed prototypes also had both a `name` and `seedName` field. The name field went unused, and was almost always equal either to the prototype Id or the `seedName`, so I just removed that and renamed `seedName` to just `name`.

:cl:
- fix: Fixed botany seeds created using the seed extractor not inheriting traits.

### CODE REVIEW

**Comment by Elijahrane** [comment]
Closes https://github.com/space-wizards/space-station-14/issues/5969

File: Content.Server/Botany/Components/ProduceComponent.cs [hunk 1]
```diff
@@ -11,5 +9,15 @@ public sealed class ProduceComponent : Component
...
+    /// <summary>
+    ///     Name of a base seed prototype that this produce can spawn.
+    /// </summary>
+    [DataField("seed", customTypeSerializer: typeof(PrototypeIdSerializer<SeedPrototype>))]
+    public readonly string? SeedName;
+
+    /// <summary>
+    ///     Uid of a modified seed prototype that this produce can spawn. Takes priority over <see cref="SeedName"/>.
+    /// </summary>
+    [DataField("SeedUid")]
```
Comment (mirrorcult) [inline Content.Server/Botany/Components/ProduceComponent.cs:21]:
```suggestion
    [DataField("seedUid")]
```

File: Content.Server/Botany/Systems/SeedExtractorSystem.cs [hunk 2]
```diff
@@ -32,8 +32,14 @@ private void OnInteractUsing(EntityUid uid, SeedExtractorComponent component, In
...
-            if (!_prototypeManager.TryIndex<SeedPrototype>(produce.SeedName, out var seed))
-                return;
+            SeedPrototype? seed;
+            // try get seed from seed database
+            if (produce.SeedUid == null || !_botanySystem.Seeds.TryGetValue(produce.SeedUid.Value, out seed))
+            {
+                // try get seed from base prototype
+                if (produce.SeedName == null || !_prototypeManager.TryIndex(produce.SeedName, out seed))
+                    return;
+            }
```
Comment (mirrorcult) [inline Content.Server/Botany/Systems/SeedExtractorSystem.cs:42]: can this get pulled out into a method somewhere, its used in 3 diff places

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/Botany/Systems/SeedExtractorSystem.cs (hunk 2 above)
Comment (ElectroJr) [reply Content.Server/Botany/Systems/SeedExtractorSystem.cs]: Moved into a function

**Review by mirrorcult — APPROVED** [review]

**Comment by mirrorcult** [comment]
conflicts then merge

**Pull request merged**